src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/LocalCIDockerService.java

```diff
@@ -290,4 +299,31 @@ public void deleteOldDockerImages() {
             }
         }
     }
+
+    /**
+     * Periodically checks for available disk space and triggers the cleanup of old Docker images if the available space falls below a specified threshold.
+     * The threshold is configurable via the 'artemis.continuous-integration.image-cleanup.disk-space-threshold-mb' property in the application settings.
+     */
+
+    @Scheduled(fixedRateString = "${artemis.continuous-integration.image-cleanup.disk-space-check-interval-ms:3600000}")
+    public void CheckUsableDiskSpaceThenCleanUp() {
```

Comment:
Is there a reason why the first letter of the method is a capital letter?

Reply:
No, I need to get used to Java again :')